server/protocol/qdatastream.py

```diff
@@ -95,7 +95,11 @@ async def read_message(self):
         if action in ("PING", "PONG"):
             return {"command": action.lower()}
 
-        message = json.loads(action)
+        try:
+            message = json.loads(action)
+        except json.decoder.JSONDecodeError as err:
+            raise json.decoder.JSONDecodeError(
```

Comment:
Need to copy all the fields from the error.
```python3
>>> e = JSONDecodeError("foobar")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() missing 2 required positional arguments: 'doc' and 'pos'
```

Reply:
sorry bout that, new commit in this PR fixes this